source/browseMode.py

```diff
@@ -1436,12 +1444,13 @@ def event_gainFocus(self, obj, nextHandler):
 				self._replayFocusEnteredEvents()
 			return nextHandler()
 
+		try:
+			lastFocusInfo=self.makeTextInfo(lastFocusObj) if lastFocusObj else None
+		except LookupError:
+			lastFocusInfo=None
 		#We only want to update the caret and speak the field if we're not in the same one as before
-		caretInfo=self.makeTextInfo(textInfos.POSITION_CARET)
-		# Expand to one character, as isOverlapping() doesn't treat, for example, (4,4) and (4,5) as overlapping.
-		caretInfo.expand(textInfos.UNIT_CHARACTER)
-		if not self._hadFirstGainFocus or not focusInfo.isOverlapping(caretInfo):
-			# The virtual caret is not within the focus node.
+		if not self._hadFirstGainFocus or not lastFocusInfo or focusInfo.compareEndPoints(lastFocusInfo,"startToStart")!=0 or focusInfo.compareEndPoints(lastFocusInfo,"endToEnd")>0:

```

Comment:
Like your other suggestion above, I think I'd prefer to just split the line up and add comments for each  comparison. A method seems a bit strange as  it would be only acting upon its arguments with no state.